### PR TITLE
Add getters for private default Headers and authentications

### PR DESCRIPTION
**Pull request by NANASHI0X74**
Rationale for this is: At the moment I'm in a situation where code generation (or rather the scheme generation for our backend code) doesn't generate code that works with our backend, so I need to do one specific request by hand. I'd like to still use the same default headers and authentications that the Apiclient uses though.

This is a really minor change, so I thought I'd show you the code before talking. Submitting this as draft so I can do the build and sample generation later

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
Makes sense, follow the PR check list and regenerate the samples. Otherwise it won't build.

I would recommend returning immutable copies of the collections in the getters to avoid the problems that you added as documentation.

**Comment by NANASHI0X74** [comment]
returning an immutable list for the authentications sounds like a good idea.
If I were to add entries to the default header map, those would get picked up though, right? I think it would have benefits to leave that mutable, for example I could use addAll() to add multiple headers at once and the like

**Comment by kuhnroyal** [comment]
Hmm are you sure this is the correct template? The `dart` template folder is unmaintained.

**Comment by NANASHI0X74** [comment]
so, I finally got round to doing this stuff. Thanks btw, I did mistakenly use the wrong template folder.
I took the liberty btw to also make the shebangs in two of the scripts portable (i.e. use /usr/bin/env bash) because the hardcoded /bin/bash actually caused problems on my nixOS machine.
Now, to mention the rest of the technical committee...
@swipesight @jaumard @josh-burton @amondnet @sbu-WBT @agilob

File: bin/utils/export_docs_generators.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (agilob) [inline bin/utils/export_docs_generators.sh:1]: Out of scope for this PR

File: bin/utils/export_generator.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (agilob) [inline bin/utils/export_generator.sh:1]: Out of scope for this PR

**Comment by agilob** [comment]
I don't understand why should none be added or deleted. If I want to add another default header conditionally from my code, I should be able to. Should your code copy the headers before any other action rather than making it immutable for everyone by force?

File: bin/utils/export_docs_generators.sh (hunk 1 above)
Comment (agilob) [reply bin/utils/export_docs_generators.sh:1]: If you want to add this in this PR, for git-log clarity it should come in own separate commit.

**Comment by NANASHI0X74** [comment]
I would actually advocate for leaving the returned headermap mutable. For the Authentications it definitely makes sense to return them immutably since the authentications are defined in the spec and the generated code sort of enforces those by only respecting authentications on certain keys in the map from what I understand, but leaving the headermap mutable has the benefit that you can use the full dart collection api to mutate it if you want and not be constrained to only adding one header at a time via a function call. I don't really see any downside to that either...

**Comment by kuhnroyal** [comment]
Fine by me

**Review by kuhnroyal — APPROVED** [review]

**Comment by wing328** [comment]
CircleCI failure not related to this change.

**Comment by kuhnroyal** [comment]
Broken master build fixed in https://github.com/OpenAPITools/openapi-generator/pull/8581

**Pull request merged**